docs/rag/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,438 @@
+
+For the Taxonomy path (no Model Training):
+
+```bash
+ilab rag convert --output /path/to/processed/folder
+```
+
+For the Plug-and-Play RAG path:
+
+```bash
+ilab rag convert --input /path/to/docs/folder --output /path/to/processed/folder
+```
+
+#### Processing-Command Purpose
+
+Applies the docling transformation to the customer documents.
+
+* Original documents are located in the `/path/to/docs/folder` input folder or in the taxonomy knowledge branch.
+  * In the latter case, the input documents are the knowledge documents retrieved from the installed taxonomy repository
+    according to the [SDG diff strategy][sdg-diff-strategy], e.g. `the new or changed YAMLs using git diff, including untracked files`.
+* Processed artifacts are stored under `/path/to/processed/folder`.
+
+***Notes**:
+
+* In alignment with the current SDG implementation, the `--input` folder will not be navigated recursively. Only files located at the root
+  level of the specified folder will be considered. The same principle applies to all other options outlined below.
+* To ensure consistency and avoid issues with document versioning or outdated artifacts, the destination folder will be cleared
+  before execution. This ensures it contains only the artifacts generated from the most recent run.
+
+The transformation is based on the latest version of the docling `DocumentConverter` (v2).
+The alternative to adopt the `instructlab-sdg` modules (e.g. the initial step of the `ilab data generate` pipeline) has been
+discarded because it generates documents according to the so-called legacy docling schema.
+
+#### Processing-Usage
+
+The generated artifacts can later be used to generate and ingest the embeddings into a vector database.
+
+### 2.3 Document Processing Pipeline Options
+
+**Note**: The `--help` option will be aware of the `rag` command group only if `ILAB_DEV_PREVIEW` environment variable is set to `true`.
+
+```bash
+% ilab rag convert --help
+Usage: ilab rag convert [OPTIONS]
+
+  The document processing pipeline for retrieval augmented generation
+
+Options:
+  --input DIRECTORY     The folder with user documents to process. In case
+                        it's missing, the knowledge taxonomy files will be
+                        processed instead.
+  --taxonomy-path PATH  Directory where taxonomy is stored and accessed from.
+  --taxonomy-base TEXT  Branch of taxonomy used to calculate diff against.
+  --output DIRECTORY    Directory where processed docs are stored.
+  --help                Show this message and exit.
+```
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Location folder of user documents. In case it's missing, the taxonomy is navigated to look for updated knowledge documents.|  | `--input` | `ILAB_PROCESS_INPUT` |
+| Location folder of processed documents. |  | `--ouput` | `ILAB_PROCESS_OUTPUT` |
```

Comment:
@anastasds , should ILAB_PROCESS_INPUT and ILAB_PROCESS_OUTPUT be ILAB_CONVERT_INPUT and ILAB_CONVERT_OUTPUT, since process was renamed to convert?

Reply:
Yes, missed that, thanks - submitted fix in https://github.com/dmartinol/dev-docs/pull/4

Reply:
Merged, thanks!